database/usage_tracker.py

```
@@ -244,26 +200,33 @@ def track_responses_create(self, model: str, operation_type: str, **kwargs) -> A
                     response_details={'usage': usage_data} if success else None
                 )
                 
-                # Log the usage
+                # Log the usage with LiteLLM cost data and Phase 2 enhancements  
                 if success:
+                    # Check if local model was used
+                    cost_source = getattr(response, '_hidden_params', {}).get('cost_source', 'unknown')
```

Ruff (0.12.2)

Local variable `cost_source` is assigned to but never used

Remove assignment to unused variable `cost_source`

(F841)